src/invocation/InvocationService.ts

```diff
@@ -21,12 +23,16 @@ class InvocationService {
         this.smartRoutingEnabled = hazelcastClient.getConfig().networkConfig.smartRouting;
     }
 
-    invokeOnConnection(connection: ClientConnection, request: ClientMessage): Q.Promise<ClientMessage> {
+    invokeOnConnection(connection: ClientConnection, request: ClientMessage, handler: (...args: any[]) => any = null)
```

Comment:
actually I will refactor this part and rebase this PR if we merge https://github.com/hazelcast-incubator/hazelcast-nodejs-client/pull/42 . When we merge that PR we can create ListenerInvocation and attach the handler to Invocation like in https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/invocation.py#L60